src/Workspaces/Core/Portable/Utilities/AsyncLazy`1.cs

```diff
@@ -294,6 +294,13 @@ private Request CreateNewRequest_NoLock()
 
         public override Task<T> GetValueAsync(CancellationToken cancellationToken)
```

Comment:
My understanding is that, realisitically, there shouldn't be a perf hit for cancellation checks. That's what we've observed when measuring ourselves, and is hte advice given when we've reached out about this.

Effectively, the cancellaiton check is branch predicted away for all the times that it is false.